lisa/features/disks.py

```diff
@@ -64,6 +67,25 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
     def get_resource_disk_mount_point(self) -> str:
         raise NotImplementedError
 
+    # Get boot partition of VM by looking for "/boot" and "/boot/efi"
+    def get_os_boot_partition(self) -> Optional[PartitionInfo]:
+        partition_info = self._node.tools[Mount].get_partition_info()
+        boot_partition: Optional[PartitionInfo] = None
+        for partition in partition_info:
+            if partition.mount_point.startswith("/boot"):
+                boot_partition = partition
+                break
+        return boot_partition
+
+    # Get disk controller type from the VM by checking the boot partition
+    def get_os_disk_controller_type(self) -> schema.DiskControllerType:
+        boot_partition = self.get_os_boot_partition()
+        assert boot_partition, "'boot_partition' must not be 'None'"
+        if boot_partition.disk == "nvme":
+            return schema.DiskControllerType.NVME
+        else:
```

Comment:
Only set to scsi, if it is. For other cases, raise exception unless we know how it happens.

Reply:
Added condition check for SCSI and exception for others.